app/models/locations/map.rb

```diff
@@ -0,0 +1,59 @@
+module Locations
+  class Map
+    include ActiveModel::Model
+    validates :loc, presence: true
+
+    def initialize(params = {})
+      @@locator_url = "http://library.princeton.edu/locator/index.php?"
+      @@stackmap_url = "http://princeton.stackmap.com/view/?"
+      @params = params
```

Comment:
Named parameters usually feel a little more useful here - or set up the known quantities you need in the meantime, via something like `@loc = params.fetch(:loc)`